src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Tests/Functions/ProcessCompletionOfNextDataSetVersionImportFunctionTests.cs

```diff
@@ -117,6 +124,3168 @@ private static Mock<TaskOrchestrationContext> DefaultMockOrchestrationContext(Gu
+                .ToDictionary(m => m.PublicId);
+
+            var newIndicatorMetas = newVersion.IndicatorMetas
+                .ToDictionary(m => m.PublicId);
+
+            // The changes should be inserted into each database table ordered alphabetically by 'Label'.
+            // They should also be ordered such that all additions come last.
+
+            // Therefore, the expected order of Indicator changes are (as per their Public IDs):
+            // Sqid index 2 deleted
+            // Sqid index 4 changed
+            // Sqid index 3 changed
+            // Sqid index 1 deleted
+            // Sqid index 6 added
+            // Sqid index 5 added
+
+            AssertIndicatorDeleted(
+                expectedIndicatorMeta: oldIndicatorMetas[SqidEncoder.Encode(2)],
+                change: indicatorMetaChanges[0]);
+            AssertIndicatorChanged(
+                expectedOldIndicatorMeta: oldIndicatorMetas[SqidEncoder.Encode(4)],
+                expectedNewIndicatorMeta: newIndicatorMetas[SqidEncoder.Encode(4)],
+                change: indicatorMetaChanges[1]);
+            AssertIndicatorChanged(
+                expectedOldIndicatorMeta: oldIndicatorMetas[SqidEncoder.Encode(3)],
+                expectedNewIndicatorMeta: newIndicatorMetas[SqidEncoder.Encode(3)],
+                change: indicatorMetaChanges[2]);
+            AssertIndicatorDeleted(
+                expectedIndicatorMeta: oldIndicatorMetas[SqidEncoder.Encode(1)],
+                change: indicatorMetaChanges[3]);
+            AssertIndicatorAdded(
+                expectedIndicatorMeta: newIndicatorMetas[SqidEncoder.Encode(6)],
+                change: indicatorMetaChanges[4]);
+            AssertIndicatorAdded(
+                expectedIndicatorMeta: newIndicatorMetas[SqidEncoder.Encode(5)],
+                change: indicatorMetaChanges[5]);
+        }
+
+        private static void AssertSingleIndicatorDeleted(
+            IReadOnlyList<IndicatorMetaChange> changes,
+            IndicatorMeta expectedIndicatorMeta)
+        {
+            Assert.Single(changes,
+                c => c.PreviousStateId == expectedIndicatorMeta.Id
+                     && c.CurrentStateId is null);
+        }
+
+        private static void AssertSingleIndicatorAdded(
+            IReadOnlyList<IndicatorMetaChange> changes,
+            IndicatorMeta expectedIndicatorMeta)
+        {
+            Assert.Single(changes,
+                c => c.PreviousStateId is null
+                     && c.CurrentStateId == expectedIndicatorMeta.Id);
+        }
+
+        private static void AssertSingleIndicatorChanged(
+            IReadOnlyList<IndicatorMetaChange> changes,
+            IndicatorMeta expectedOldIndicatorMeta,
+            IndicatorMeta expectedNewIndicatorMeta)
+        {
+            Assert.Single(changes,
+                c => c.PreviousStateId == expectedOldIndicatorMeta.Id
+                     && c.CurrentStateId == expectedNewIndicatorMeta.Id);
+        }
```

Comment:
As per other comments - not sure we need these assertions when we have the more precise ones below.

Reply:
See other comment

Reply:
Similar to other comment - https://github.com/dfe-analytical-services/explore-education-statistics/pull/5227#discussion_r1790259731